<!DOCTYPE html>
<html>
  <head>
    
      <title>James Madison University</title>
    
    <link rel="shortcut icon" type="image/x-icon" href="./favicon.ico" />
<link href="./css/bootstrap/bootstrap.css" rel="stylesheet" />
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="./css/bootstrap/bootstrap-responsive.css" rel="stylesheet" />
<link rel="stylesheet" type="text/css" href="./css/swc.css" />
<link rel="stylesheet" type="text/css" href="./css/swc-bootstrap.css" />
<meta charset="UTF-8" />
<meta http-equiv="last-modified" content="" />
<!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

    <meta name="venue" content="James Madison University" />
    <meta name="date" content="July 22-23, 2015" />
    <meta name="startdate" content="2015-07-22" />
    <meta name="enddate" content="2015-07-23" />
  </head>
  <body>
    <div class="container">
      <div class="span10 offset1">
        <div class="banner">
  <a href="http://datacarpentry.org" title="Data Carpentry">
    <img alt="Data Carpentry banner" src="./img/DC1_logo_small.png" height="50px" />
  </a>
</div>


        <div class="hero-unit">
          <div class="row-fluid">
            <div class="span10">
	      <h2>A Genomics Data Carpentry Workshop</h2>
              <h2>James Madison University</h2>
              Bioscience Building, room 2007<br/>
              July 22-23, 2015
              <br/>9:00 am - 5:00 pm
	    </div>
          </div>
        </div>

        <!--
    Edit the values in the parameter block above to be appropriate for your bootcamp.
    Please use three-letter month names for the 'humandate' field.
-->

<!--
    This block includes the Eventbrite registration widget if 'eventbrite' has been set in the header.
-->


<h2>General Information</h2>
<!--
<p>
    More information about this workshop is available at: <a href=https://github.com/datacarpentry/2015-05-30-HASTAC/wiki>https://github.com/datacarpentry/2015-05-30-HASTAC/wiki</a>
</p>
-->
<p>
Data Carpentry workshops are for any researcher who has data they want to analyze
, and no prior computational experience is required. This hands-on workshop 
teaches basic concepts, skills and tools for working more effectively with 
data. 

<p><b>The focus of this workshop will be on working with genomics data
and data management and analysis for genomics research.</b> 
We will cover metadata organization in spreadsheets, connecting to 
and using cloud computing, the command line for 
sequence quality control and bioinformatics workflows,
and R for data analysis and visualization. 
We will not be teaching any particular bioinformatics tools, but the 
foundational skills that will allow you to conduct any analysis and
analyze the output of a genomics pipeline. 
<p>
Participants should bring their laptops and plan to participate 
actively. By the end of the workshop learners should be able to more effectively 
manage and analyze data and be able to apply the tools and approaches 
directly to their ongoing research. 
<p>
Data Carpentry's aim is to teach researchers basic concepts, skills,
and tools for working with data so that they can get more done in less
time, and with less pain.
</p>
<p>
<!--
Preliminary schedule:
<ul>
<li>  Day 1 morning: Planning a genomics project and metadata organization
<li>  Day 1 afternoon: Data analysis and visualization in R
<li>  Day 2 morning: The command line for bioinformatics
<li>  Day 2 afternoon: Using cloud computing for bioinformatics
</ul>

-->


</p>

<p>Updates will be posted to this website as they become available.</p>

<p>The etherpad for this workshop can be found <a href = "https://etherpad.mozilla.org/2015-07-22-JamesMadison">here</a>.</p>


<!--
<p>Data files for the workshop are available at <a href = "/2015-01-15-cornell/data/biology/">the following link</a></p>


<p>A static copy of the etherpad as of 5:30 on Friday after the workshop ended is <a href ="etherpad-snapshot-530-Friday.html">here</a></p>
-->

<!-- This block displays the instructors' names if they are available. -->


<p>
  <strong>Organizer:</strong>
  James Herrick (James Madison), Ray Enke (James Madison), Steve Cresawn (James Madison)
</p>
<p>
  <strong>Instructors:</strong>
  Stephen Turner (University of Virginia), Amanda Charbonneau (Michigan State University)
</p>
<p>
  <strong>Assistants:</strong>
  Pete Nagraj (University of Virginia)



<!--
    Modify this block to reflect the target audience for your bootcamp.
    In particular, if it is only open to people from a particular institution,
    or if specialized prerequisite knowledge is required, please mention that.
-->
<p>
  <strong>Who:</strong>
  The course is aimed at faculty, research staff, postdocs, graduate students, advanced undergraduates, and other researchers in any field. No prior computational experience is required. 
</p>

<!--
    This block displays the address and links to a map showing directions.
-->

<p>
  <strong>Where:</strong>
  Bioscience Building, room 2007.
  Get directions with
  <a href="http://www.openstreetmap.org/?mlat=38.4338888&mlon=-78.8605821&zoom=16">OpenStreetMap</a>
  or
  <a href="http://maps.google.com/maps?q=38.4338888,-78.8605821">Google Maps</a>. 
</p>


<!--
    Modify the block below if there are any special requirements.
-->
<p>

<p>
  <strong>Requirements:</strong>
Data Carpentry's teaching is hands-on, so participants are encouraged to bring in and use their own laptops to insure the proper setup of tools for an efficient workflow once you leave the workshop.  (We will provide instructions on setting up the required software several days in advance.). <em> There are no pre-requisites, and we will assume no prior knowledge about the tools.</em> Participants are required to abide by Software Carpentry's
<a href="http://software-carpentry.org/conduct.html">Code of Conduct</a>.
</p>

<!--
    This block automatically inserts a contact email address if one has been specified for the page.
    If one hasn't, this block inserts the generic contact address for Software Carpentry.
-->
<p>
  <strong>Contact</strong>:
  Please email
  
    <a href='mailto:tkteal@datacarpentry.org'>tkteal@datacarpentry.org</a>
  
  for questions and information not covered here.
</p>

<p><strong>Twitter</strong>: #datacarpentry</p> @datacarpentry

<a name="setup"></a>

<!--
    Edit this block to show the syllabus and schedule for your bootcamp.
-->

<h2>Schedule</h2>
<table class="table table-striped">
  <tr> <td>Wednesday</td>  <td>08:30</td>  <td>Breakfast, and optional help with setup prior to starting</td> </tr>
  <tr> <td></td>        <td>09:00</td>       <td> Data organization and management (Amanda)</td> </tr>
  <tr> <td></td>        <td></td>       <td>Refreshments will be served around 10:30.</td> </tr>
  <tr> <td></td>        <td>10:45</td>       <td> Working with genomics file types (Amanda)</td> </tr>  
  <tr> <td></td>        <td>12:30</td>  <td>Lunch break </td> </tr>
  <tr> <td></td>        <td>13:30</td>  <td> Introduction to R (Stephen)</td> </tr>
  <tr> <td></td>        <td></td>       <td>Refreshments will be served around 15:00.</td> </tr>
  <tr> <td></td>        <td>15:15</td>  <td> Advanced manipulation & visualization with R (Stephen) </td> </tr>  
  <tr> <td></td>        <td>16:30</td>  <td>Wrap-up</td> </tr>
  <tr> <td></td>        <td>17:00</td>  <td>Evening Mixer (5pm to 7pm)</td> </tr>
  <tr> <td>Thursday</td>  <td>08:30</td>  <td>Breakfast, and optional help with setup prior to starting</td> </tr>
  <tr> <td></td>        <td>09:00</td>  <td> Introduction to command line (Amanda)</td> </tr>
  <tr> <td></td>        <td></td>       <td>Refreshments will be served around 10:30.</td> </tr>
  <tr> <td></td>        <td>10:45</td>       <td>Wrangling and processing genomic data-Advanced shell (Amanda)</td> </tr>  
  <tr> <td></td>        <td>12:30</td>  <td>Lunch break </td> </tr>
  <tr> <td></td>        <td>13:30</td>  <td> Introduction to cloud computing (Stephen) </td> </tr>
  <tr> <td></td>        <td></td>       <td>Refreshments will be served around 15:00.</td> </tr>
  <tr> <td></td>        <td>15:15</td>  <td> Cloud computing: example analysis & data transfer (Stephen) </td> </tr>    
  <tr> <td></td>        <td>16:30</td>  <td>Wrap-up</td> </tr>
  <tr> <td></td>        <td>17:00</td>  <td>Evening Mixer (5pm to 7pm)</td> </tr>
</table>

<hr/>




<h2>Setup</h2>

<h3>R + Rstudio</h3>

<!DOCTYPE html><html><head><meta charset="utf-8"><title>setup-r</title></head><body><article class="markdown-body"><p><strong>Note:</strong> R and RStudio are separate downloads and installations. <strong>R</strong> is the underlying statistical computing environment, but using R alone is no fun. <strong>RStudio</strong> is a graphical integrated development environment that makes using R much easier. You need R installed before you install RStudio.</p>
<ol>
<li><strong>Download the gapminder data.</strong> <a href="http://bioconnector.org/data/gapminder.csv">Click here to download the data</a> that we will use for this section. Save it somewhere you&rsquo;ll remember.</li>
<li><strong>Install R.</strong> You&rsquo;ll need R version 3.2.0 or higher. Download and install R for <a href="http://cran.r-project.org/bin/windows/base/">Windows</a> or <a href="http://cran.r-project.org/bin/macosx/">Mac OS X</a> (download the latest R-3.x.x.pkg file for your appropriate version of OS X).</li>
<li><strong>Install RStudio.</strong> Download and install the latest stable version of <a href="http://www.rstudio.com/products/rstudio/download/">RStudio Desktop</a>.</li>
<li><strong>Install R packages.</strong> Launch RStudio (RStudio, <em>not R itself</em>). Ensure that you have internet access, then enter the following commands into the <strong>Console</strong> panel (usually the lower-left panel, by default). Note that these commands are case-sensitive. At any point (especially if you&rsquo;ve used R/Bioconductor in the past), R may ask you if you want to update any old packages by asking <code>Update all/some/none? [a/s/n]:</code>. If you see this, type <code>a</code> at the propt and hit <code>Enter</code> to update any old packages. If you&rsquo;re using a Windows machine you might get some errors about not having permission to modify the existing libraries &ndash; don&rsquo;t worry about this message. You can avoid this error altogether by running RStudio as an administrator.</li>
<li><strong>Download the data.</strong> <a href="etc/gapminder.csv">Click here to download the data</a> that we will use for the R lesson. Save it somewhere easy to remember and fine e.g., your Desktop. <em>(Alternative data source: <a href="http://bioconnector.org/data/gapminder.csv">http://bioconnector.org/data/gapminder.csv</a>)</em>.</li>
</ol>
<pre><code># Install packages from CRAN
install.packages(&quot;dplyr&quot;)
install.packages(&quot;ggplot2&quot;)
install.packages(&quot;tidyr&quot;)

# Install Bioconductor base
source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite()
</code></pre>

<p>You can check that you&rsquo;ve installed everything correctly by closing and reopening RStudio and entering the following commands at the console window:</p>
<pre><code>library(dplyr)
library(ggplot2)
library(tidyr)
library(Biobase)
</code></pre>

<p>These commands may produce some notes or other output, but as long as they work without an error message, you&rsquo;re good to go. If you get a message that says something like: <code>Error in library(packageName) : there is no package called 'packageName'</code>, then the required packages did not install correctly.</p></article></body></html>

<h3>Shell / cloud</h3>

<!DOCTYPE html><html><head><meta charset="utf-8"><title>setup-shell</title></head><body><article class="markdown-body"><p>Most bioinformatics is done on a computer running a Linux/UNIX operating system. In this workshop we will be doing data analysis on a remote linux server rather than on our own laptops. To do that we need: (1) a remote computer set up with all the software we&rsquo;ll need, (2) a way to connect to that computer, and (3) a way to transfer files to and from that computer.</p>
<p>Since most of us don&rsquo;t have our own Linux server running somewhere, we&rsquo;ll <em>rent</em> a server from Amazon for the duration of this course.</p>
<!--

**Create AWS account**

First, create an Amazon Web Services account: <http://aws.amazon.com/>. Make sure to register for a Basic (Free) account. You will be required to enter a credit card and billing information -- don't worry, I have free use vouchers for you so you will not be charged. You will need to verify a phone number before you can start using AWS. Note that your Amazon.com account is not connected to your Amazon Web Services account. They are two separate entities with different login and billing information.

Once you have your AWS account set up and can successfully log in to [console.aws.amazon.com](https://console.aws.amazon.com/), email Stephen Turner (<a href="http://www.google.com/recaptcha/mailhide/d?k=01uXi4zl-bIdygzSeXF4649A==&amp;c=_81hv-sTQvJ9rjELjZNDJeAXTvLvkpfD9KEuItpEHTE=" onclick="window.open('http://www.google.com/recaptcha/mailhide/d?k\07501uXi4zl-bIdygzSeXF4649A\75\75\46c\75_81hv-sTQvJ9rjELjZNDJeAXTvLvkpfD9KEuItpEHTE\075', '', 'toolbar=0,scrollbars=0,location=0,statusbar=0,menubar=0,resizable=0,width=500,height=300'); return false;" title="Reveal this e-mail address">sd...</a>@virginia.edu) to obtain a voucher to be able to use AWS for free during and after our course. Use the subject line "RNA-SEQ COURSE AWS VOUCHER" in your email to me. Once you have your voucher, return to the AWS console, click your name at the top right, click "Billing & Cost Management", then on the left, click "Credits". Redeem the promo code I sent you -- this credit will buy enough compute time to complete this workshop and for several future RNA-seq analyses.

If you're interested in trying out EC2 prior to the workshop, watch [this short video](http://youtu.be/SKM0BB0F02Q) to learn how to launch your first instance (and make sure to stop the instance after you're done). Whether you do this or not, *be sure to stop or terminate any running EC2 instances when you are done with them*. After the course, you may deactivate your AWS account if you wish under the "My Account" settings in the AWS console.

-->

<ul>
<li><strong>(<em>All participants</em>) Download a file transfer program.</strong> Download and install Cyberduck (free): <a href="https://cyberduck.io">https://cyberduck.io</a>. We will use this transfer files back and forth between our local laptops and our remote linux server running on Amazon. Note: If using a Mac, download from the website above (free), <em>not</em> from the Mac App Store (paid). </li>
<li><strong>(<em>All participants</em>) Download a text editor.</strong> We may wish to view and/or edit plain text files. To do this, let&rsquo;s use a better alternative to the built-in Notepad (Windows) or TextEdit (Mac). Download and install Sublime Text (works for Windows and Mac): <a href="http://www.sublimetext.com/">http://www.sublimetext.com/</a>. </li>
<li><strong><em>For Windows users only</em></strong>:<ul>
<li><strong>Install git-bash</strong>. Download and run <a href="http://msysgit.github.io/">the installer here</a>. This will install a &ldquo;real&rdquo; command line on a Windows machine.</li>
<li><strong>Install the Software Carpentry Installer</strong>. <em>After</em> installing git-bash, <a href="http://files.software-carpentry.org/SWCarpentryInstaller.exe">download and install the software carpentry installer here</a>.</li>
<li><strong>Download a terminal emulator.</strong> <ul>
<li><a href="http://the.earth.li/~sgtatham/putty/latest/x86/putty.exe">Download the latest version of PuTTY here</a>. Ensure that you can run this .exe file (no installation necessary). This is your terminal emulator.</li>
<li><a href="http://the.earth.li/~sgtatham/putty/latest/x86/puttygen.exe">Download puttygen here</a>. This will allow you to generate a key used to log in to your cloud computing instance.</li>
</ul>
</li>
</ul>
</li>
<li><strong>Bookmark these links</strong>: These resources will be useful for logging in and transferring data to your instance.<ul>
<li><a href="http://angus.readthedocs.org/en/2014/amazon/transfer-files-between-instance.html">Transferring data between your computer and your EC2 instance</a></li>
<li><a href="http://angus.readthedocs.org/en/2014/amazon/log-in-with-ssh-win.html">Logging on to your instance for windows</a></li>
<li><a href="http://angus.readthedocs.org/en/2014/amazon/log-in-with-ssh-mac.html">Logging on to your instance for Mac</a></li>
</ul>
</li>
</ul></article></body></html>



<h2>Acknowledgements &amp; Support</h2>
<!--
<div class="pull-right" style="max-width: 320px; padding-left: 6px;">
</div>
-->
<p>
Data Carpentry is supported by the <a href=http://http://www.moore.org/>Gordon and Betty Moore Foundation</a> and a partnership of several NSF-funded <a href="http://www.nsf.gov/dir/index.jsp?org=BIO" target="_blank">BIO</a> Centers (<a href="http://nescent.org" target="_blank">NESCent</a>, <a href="http://iplantcollaborative.org" target="_blank">iPlant</a>, <a href="http://idigbio.org" target="_blank">iDigBio</a>, <a href="http://beacon-center.org/" target="_blank">BEACON</a> and <a href="http://sesync.org" target="_blank">SESYNC</a>) and <a href="http://software-carpentry.org" target="_blank">Software Carpentry</a>. 
The structure and objectives of the curriculum as well as the teaching style are informed by <a href="http://software-carpentry.org" target="_blank">Software Carpentry</a>.
</p>




<!--
    Edit the setup instructions in _includes/setup.html to reflect your bootcamp.
    (In particular, most bootcamps teach either Python or R, not both.)




<h2>Additional Resources</h2>

<h3>shell</h3>
<ul>
<li><a href=http://fosswire.com/post/2007/08/unixlinux-command-cheat-sheet/>Unix/Linux Command Reference</a>
<li><a href=https://github.com/swcarpentry/boot-camps/blob/master/shell/shell_cheatsheet.md
>Shell cheat sheet</a>
<li><a href=http://software-carpentry.org/v4/shell/index.html>Software Carpentry shell lessons</a>
</ul>

<h3>R</h3>

<b>Where to learn more about R</b>
<ul>
<li><a href=http://www.statmethods.net/>http://www.statmethods.net/</a> - good for data organization, basics stats and graphs
<li><a href=http://www.gardenersown.co.uk/Education/Lectures/R/anova.htm>http://www.gardenersown.co.uk/Education/Lectures/R/anova.htm</a> - basic parametric and non-parametric stats
<li><a href=http://www.cyclismo.org/tutorial/R/index.html>http://www.cyclismo.org/tutorial/R/index.html</a> - R tutorial
<li><a href=http://www.amazon.com/R-Action-Robert-Kabacoff/dp/1935182390>R in Action</a> - good book as an R reference
<li><a href=http://www.twotorials.com/>http://www.twotorials.com/</a>
<li><a href=http://www.r-bloggers.com/>http://www.r-bloggers.com/</a>
<li><a href=http://tryr.codeschool.com/>http://tryr.codeschool.com/</a>
<li><a href=http://adv-r.had.co.nz/>Advanced R Programming by Hadley Wickham</a>
<li><a href=http://www.computerworld.com/s/article/9239625/Beginner_s_guide_to_R_Introduction>Beginner's Guide to R from Computer World</a>
<li><a href=http://www.scoop.it/t/r-for-journalists>R for Journalists</a>
<li><a href=http://www.r-bloggers.com/>R Bloggers</a>
<li><a href=http://www.inside-r.org/>inside-R</a> 
<li><a href=http://ropensci.org/>rOpenSci</a>
</ul>

<p>
<b>Plotting in R</b>
<ul>
<li><a href=http://www.harding.edu/fmccown/r/>http://www.harding.edu/fmccown/r/</a> - Very simple graphs
<li><a href=https://storify.com/tracykteal/r-galleries>A variety of R gallery recommendations</a>
<li><a href=http://docs.ggplot2.org/>ggplot gallery</a> - extensive and comprehensive; a great resource
<li><a href=http://www.amazon.com/R-Graphics-Cookbook-Winston-Chang-ebook/dp/B00AJ5X7W4/>R Graphics Cookbook</a> - highly recommended
<li><a href=http://www.cookbook-r.com/Graphs/>Some of R Graphics Cookbook plots</a> - a set of some of the plots from the R Graphics Cookbook by Winston Chang
<li><a href=http://rgm3.lab.nig.ac.jp/RGM/R_image_list?page=665&init=true>R Graphical Manual</a> - plots from apparently every R CRAN package
</ul>
-->
<!--
<h2>Setup</h2>

<p>
  To participate in a Data Carpentry workshop, 
  you will need working copies of the shell, R, RStudio, SQL and OpenRefine.
  Please make sure to install everything
  (or at least to download the installers)
  <em>before</em> the start of your bootcamp. 
  
Participants should bring and use their own laptops to insure the proper setup of tools for an efficient workflow once you leave the workshop.
<p>
Please follow these <a href=install.html><b>Setup Instructions</b></a>  
  
</p>
-->




        <div class="footer">
  <a class="label swc-blue-bg" href="">Facebook</a>
  <a class="label swc-blue-bg" href="">Google+</a>
  <a class="label swc-blue-bg" href="">Twitter</a>
  <a class="label swc-blue-bg" href="">GitHub</a>
  <a class="label swc-blue-bg" href="./LICENSE.html">License</a>
  <a class="bugreport label swc-blue-bg" href="mailto:?subject=bug%20in%20index.html">Bug Report</a>
</div>

      </div>
    </div>
    <!-- Le javascript
================================================== -->
<!-- Placed at the end of the document so the pages load faster -->
<script src="./js/jquery-1.9.1.min.js"></script>
<script src="./js/bootstrap/bootstrap.min.js"></script>

  </body>
</html>
